actix-rt/src/arbiter.rs

```diff
@@ -317,6 +318,30 @@ impl Arbiter {
     }
 }
 
+/// Future used for cleaning-up already finished `JoinHandle`s
+/// from the `PENDING` list so the vector doesn't grow indefinitely
+struct CleanupPending;
+
+impl Future for CleanupPending {
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        PENDING.with(move |cell| {
+            let mut pending = cell.borrow_mut();
+            let mut i = 0;
+            while i != pending.len() {
```

Comment:
Should be possible to save the index in `struct CleanupPending` to allow yielding after process X pending items.  X could be a percentage of the total number of pending tasks, instead of a fixed number, so the array's growth doesn't out run the cleanup.

Reply:
> Should be possible to save the index

Getting the vector's length and then yield doesn't seams like a good idea as the length could be changed doing this waiting time, either growing or getting smaller by a newly spawned cleanup.